planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -573,9 +576,9 @@ AvoidPointArray AvoidanceModule::calcRawShiftPointsFromObjects(
     AvoidPoint ap_return;
     ap_return.length = 0.0;
     ap_return.start_length = shift_length;
-    ap_return.start_longitudinal = o.longitudinal;
-    ap_return.end_longitudinal =
-      o.longitudinal + std::min(nominal_return_distance, return_remaining_distance);
+    ap_return.start_longitudinal = o.longitudinal + o.distance_add_to_returnpoint;
```

Comment:
@taikitanaka3 
I want to separate the longitudinal end of the starting point from the longitudinal start of the return point so that "BC" can maintain the maximum offset.
![2022-06-02 09-59-42屏幕截图](https://user-images.githubusercontent.com/102840938/171535913-319f4489-37b9-4a1a-af7f-59624c1e5896.png).

Reply:
@shulanbushangshu 
My apologies, I might misunderstood your intention. Assuming I take this [image](https://user-images.githubusercontent.com/102840938/171134680-95943418-72cc-4e34-9374-bf2d530cdf79.png) as the example, it essentially mean you're shifting the start_longitudinal pose from point B to point C. 

Is it possible if you can provide some sketch (simple is also fine) regarding maintaining the maximum offset for "BC so that we can better understand the situation?

Reply:
@zulfaqar-azmi-t4 @taikitanaka3 
Thank you for your review.Let me briefly describe the process of issue#833:
In the process of obstacle avoidance simulation, I found that the default lateral safety margin was a little large. When the intersection between obstacle and lane was very small, there would be an offset of about 1.5m, as shown in the figure below.
![large](https://user-images.githubusercontent.com/102840938/171581502-5f87836a-fbdb-47bd-95ce-e54314ece75f.png)

Therefore, I changed the default lateral safety margin and conducted simulation on this basis.There are two variables :''lateral_collision_margin" and "lateral_collision_safety_buffer".I set "lateral_collision_safety_buffer" to 0.0.
In the following simulation, issue#833 occurs, as shown below:
![2022-06-02 15-21-46屏幕截图](https://user-images.githubusercontent.com/102840938/171581642-ccf2b61f-3e75-40d4-a498-50c7ffebc64b.png)

Of course, if the lateral safety distance margin is enlarged, there may be no collision. But I think it is possible to reduce the risk of parameter adjustment by calculating the compensation amount of a longitudinal distance.According to the above situation, I calculated the compensation amount of the longitudinal distance according to the obstacles and lanes.I want to separate the longitudinal end of the starting point from the longitudinal start of the return point so that "BC" can maintain the maximum offset.
The distance is calculated as follows:
![2022-06-02 15-47-13屏幕截图](https://user-images.githubusercontent.com/102840938/171581741-993c3184-666f-49bb-b385-2fbef0d2e755.png)
In this example,the 'distance_add_to_returnpoint'='d2'+'d4'.
Calculation principle: calculate the minimum distance and maximum distance from the contour point and intersection point to the reference point. The maximum distance minus the minimum distance is used as the compensation distance.
Please consider whether my idea is necessary and correct.

Reply:
@shulanbushangshu 
About this [image](https://user-images.githubusercontent.com/102840938/171581642-ccf2b61f-3e75-40d4-a498-50c7ffebc64b.png), the result indeed seems weird. it is possible for you to reproduce them with the debug marker `publish_debug_marker=true` in the `avoidance.param.yaml` and share the image with us?

Reply:
@shulanbushangshu 
I've tested your code but it looks like something wrong
![image (6)](https://user-images.githubusercontent.com/65527974/172075731-fbf62390-9760-4897-b041-b4f764a1d07b.png)

Reply:
In your algorithm I think you need to add margin for start longitudinal too according to collision of foot print and lanelet.

Reply:
@zulfaqar-azmi-t4 @taikitanaka3 
Thank you for reviewing.
This is the debug marker:
![2022-06-06 14-30-58屏幕截图](https://user-images.githubusercontent.com/102840938/172108799-378e177e-35ec-44c7-a61d-377d8671dd38.png)
@taikitanaka3 ,you are right.I forgot to change the vertical distance of the starting point.I will fix it.
The two issues((#833,#934)) I raised are aimed at different problems.
The first is to solve the problem of early return (there will be problems if there are long obstacles).
The second is to solve the offset distance (there may be problems in calculating the offset distance in some cases) We also consider the points on the contour of the obstacle and the distance to the boundary. I think it is necessary to modify the calculation method of lateral offset distance.
To sum up, the first issue can be solved by longitudinal distance compensation; The second issue can be solved by lateral distance compensation.
Of course, the above two problems will lead to the risk of collision. In this PR, only the first issue is solved.

Reply:
@shulanbushangshu 
Good, but at first I want to merge fix PR for considering vehicle size in frenet-coordinate is it ok?
your PR looks like new feature that we haven't tested yet.(for the reason you change first avoid point and second return point according to the collision of lanelet and collision edge case is not fully defined or tested yet)  
https://github.com/autowarefoundation/autoware.universe/pull/1041#issuecomment-1147039130

Reply:
@taikitanaka3.No problem